File: SpecialityString.cpp
#include <iostream>
#include <stack>
#include <algorithm>

using namespace std;

int main(){
	int test;
	cin >> test;
	for(int k = 0; k < test; k++){
		stack<char> charStack;
		int s;
		cin >> s;
		string str;
		cin >> str;
		for(int i = 0; i < s; i++){
			char c = str[i];
			if(charStack.empty() || charStack.top() != c){
				charStack.push(c);
			}
			else{
				charStack.pop();
			}
		}
		
		if(charStack.empty()){
			cout << "YES" << endl;
		}
		else{
			cout << "NO" << endl;
		}
		
	}
}

File: SpecialityString.java
import java.util.Scanner;
import java.util.Stack;

public class Main{
	public static void main(){
		Scanner myScanner = new Scanner(System.in);
		int testCases = myScanner.nextInt();
		while(testCases-->0){
			solve(myScanner);
		}
	}
	private static void solve(Scanner getInput){
		
		int size = getInput.nextInt();
		getInput.nextLine();
		String myStr = getInput.nextLine();
		//System.out.println(myStr);
		
		Stack<Character> charStack = new Stack<>();
		
		for(int i = 0; i < myStr.length(); i++){
			if(!charStack.isEmpty() && charStack.peek() == myStr.charAt(i)){
				charStack.pop();
			}
			else{
				charStack.push(myStr.charAt(i));
			}
		}
		
		System.out.println(charStack.isEmpty() ? "YES" : "NO");
	}
}